### ISSUE TITLE
Appears in incorrect menu in debian mate

### ISSUE THREAD

**Issue by webmind** [flathub/io.github.dweymouth.supersonic#47]
In debian(bookworm) mate supersonic appears in the 'other' section of the menu instead of 'Sound & Video'. In version 0.8.2

**Comment by anarcat** [comment]
taking inkscape as an example, it looks like the issue is that the upstream "appdata" file doesn't have a `<category>` tag. example:

https://gitlab.com/inkscape/inkscape/-/blob/master/org.inkscape.Inkscape.appdata.xml.in?ref_type=heads#L29-31

```
  <categories>
    <category>Graphics</category>
  </categories>
```

so at first i thought this was a flatpak issue, but it's actually an upstream, can you open that bug report upstream instead? https://github.com/dweymouth/supersonic/issues

**Comment by anarcat** [comment]
@dweymouth i think the valid categories are in https://specifications.freedesktop.org/menu-spec/latest/apa.html seems like `Audio` is the magic string. it also says "Desktop entry must include AudioVideo as well"...

**Comment by dweymouth** [comment]
@anarcat is the Flatpak package using this desktop file? https://github.com/dweymouth/supersonic/blob/main/res/supersonic-desktop.desktop This was added by the Arch packager and AFAIK it was only used there

**Comment by dweymouth** [comment]
Added audio category [here](https://github.com/dweymouth/supersonic/commit/61f5f24c4f58f499f6429dcd9d668d4900c46001). I'm not sure how the .desktop file is generated with the Flatpak  packaging and if anything needs to be done there.

**Comment by anarcat** [comment]
i'm not sure! there's some noises in the .yml file about tweaking the desktop file:

https://github.com/flathub/io.github.dweymouth.supersonic/blob/51232b5042342893cdf2abbfbda8b79ea50b4394/io.github.dweymouth.supersonic.yml#L98

i *assume* that takes your .desktop file, but i'm actually not sure at all... in fact, i'd be surprised if it gets used at all! so to be verified.

**Comment by dweymouth** [comment]
Oh - it's using the one generated by `fyne package` for Linux. That .desktop file is very bare-bones and has no categories, nor does `fyne package` have an API to add them. So I think you need to add more `desktop-file-edit` commands to add the Audio and AudioVideo categories

**Comment by dweymouth** [comment]
Also if we push a new Flatpak build now it will pull in some unreleased new features that were merged to main. I guess I may need to rethink my branching strategy, and have a `release` branch? Anyway @webmind is it OK if this waits until 0.9.0 to be fixed? (Maybe a few weeks or a month?)

**Comment by anarcat** [comment]
here's what the .desktop file looks like here:

```
anarcat@angela:~$ cat /var/lib/flatpak/app/io.github.dweymouth.supersonic/x86_64/stable/d669f5f06d34858e84d0a5f82bd4a6944ab8e300aee30f35b4081dd21da2e3f0/files/share/applications/io.github.dweymouth.supersonic.desktop
[Desktop Entry]
Type=Application
Name=Supersonic
Exec=supersonic
Icon=io.github.dweymouth.supersonic
X-Desktop-File-Install-Version=0.27
X-Flatpak-RenamedFrom=Supersonic.desktop;
```

so it doesn't look like the one shipped by supersonic. i'll look to see how this can be fixed.

re unreleased stuff, i think this can wait for the next release to come up, otherwise we could push a beta instead...

**Comment by dweymouth** [comment]
@anarcat if you're using the .desktop file shipped by Supersonic, you might have to edit the Exec too depending on what the actual binary name is in the Flatpak. That .desktop file was created for Arch, where it's packaged as `supersonic-desktop` because there was already a `supersonic` arch package: https://aur.archlinux.org/packages/supersonic (A really niche Subsonic server implementation I didn't even know about before)

**Comment by anarcat** [comment]
> so it doesn't look like the one shipped by supersonic. i'll look to see how this can be fixed.

i have an attempt at doing this in #48, but i've closed it because it also included a pile of crap just from the beta/master sync, which is now in #49. once that's built and merge, then i'll reroll the PR.


> @anarcat if you're using the .desktop file shipped by Supersonic, you might have to edit the Exec too depending on what the actual binary name is in the Flatpak.

yeah, flatpak automatically does that rewiring for us, so no worries there.



> That .desktop file was created for Arch, where it's packaged as `supersonic-desktop` because there was already a `supersonic` arch package: https://aur.archlinux.org/packages/supersonic (A really niche Subsonic server implementation I didn't even know about before)

wtf. naming things right?

**Comment by anarcat** [comment]
https://github.com/flathub/io.github.dweymouth.supersonic/pull/50 has been pushed, a build should be available within 10-15m max. @webmind can you see if you can test the build when it's done?

**Comment by anarcat** [comment]
*finally* I got a build to work.

@webmind can you try the build generated in #50? the following command should get you the test build:

```
flatpak install --user https://dl.flathub.org/build-repo/71772/io.github.dweymouth.supersonic.flatpakref
```

if that works, i'll merge the result, which will only be in beta for now, but eventually will trickle back out to the normal release once upstream publishes a new one. :)

**Comment by webmind** [comment]
Seems to work fine in that version, no hurry for release for me.

**Comment by anarcat** [comment]
awesome, so i've merged this into beta, we just need to make sure to not forget to merge *that* into master when the next release come.

alternatively, if people are comfortable with that, i could just push beta to master as well...

**Comment by dweymouth** [comment]
> alternatively, if people are comfortable with that, i could just push beta to master as well...

It's fine by me, what was merged since 0.8.2 should be stable, it's just adding 2 new OpenSubsonic properties - comment column for tracklists and showing the release type (Album, EP, etc) on the album page header.
